doc/controller_configuration/controller_configuration_tutorial.rst

```diff
@@ -30,146 +79,409 @@ The ``controllers.yaml`` configuration file is located in the ``robot_moveit_con
+
+The only job of a controller handle allocator is to create a new instance of the controller handle. The following example implements an allocator for a custom controller handle of type ``example::controller_handle_example``:
+
+.. code-block:: c++
+
+  // declares example::controller_handle_example class
+  #include "controller_handle_example.h"
+  #include <moveit_ros_control_interface/ControllerHandle.h>
+  #include <pluginlib/class_list_macros.h>
+
+  namespace example
+  {
+  class controller_handle_allocator_example : public moveit_ros_control_interface::ControllerHandleAllocator
+  {
+  public:
+    moveit_controller_manager::MoveItControllerHandlePtr alloc(const std::string& name,
+                                                               const std::vector<std::string>& resources) override
+    {
+      return std::make_shared<controller_handle_example>(name, std::string("follow_joint_trajectory"));
+    }
+  };
+  }  // namespace example
+
+  PLUGINLIB_EXPORT_CLASS(example::controller_handle_allocator_example,
+                         moveit_ros_control_interface::ControllerHandleAllocator);
+
+
+This example controller handle allocator can be exported by creating a plugin definition file which is then referenced in the ``exports`` section of ``package.xml``:
+
+.. code-block:: XML
+
+    <library path="lib/libtrajectory_controller_example">
+        <class
+            name="example/trajectory_controller_example"
+            type="example::controller_handle_allocator_example"
+            base_class_type="moveit_ros_control_interface::ControllerHandleAllocator"
+        >
+            <description>
+                Example Controller Handle Allocator for MoveIt!
+            </description>
+        </class>
+    </library>
+
+.. note::
+  Replace ``lib/libtrajectory_controller_example`` with your library name by following the same format (prepending ``lib/lib`` to your library name)
+
+This plugin definition links the name of a controller you are integrating with MoveIt (specified by the ``name`` attribute) with the type of the allocator you implemented (specified by the ``type`` attribute), such as the one in the example above.
+
+The ``base_class_type`` must be set to ``moveit_ros_control_interface::ControllerHandleAllocator`` to make the allocator discoverable by MoveIt.
+
+The plugin definition can then be referenced in the package manifest:
+
+.. code-block:: XML
+
+    <export>
+        <!-- other exports... -->
+        <moveit_ros_control_interface plugin="${prefix}/controller_handle_allocator_plugin.xml"/>
+    </export>
+
+The translation between `moveit_msgs::RobotTrajectory <http://docs.ros.org/en/noetic/api/moveit_msgs/html/msg/RobotTrajectory.html>`_ message and the type of command supported by the controller would be done by implementing a controller handle, for example:
```

Comment:
This example (the following code snippet) simply repeats most parts of `FollowJointTrajectoryControllerHandle`.
I don't see value in repeating this code here.

Reply:
The value is that the original code for FollowJointTrajectoryControllerHandle is buried inside MoveIt. I think we should demonstrate how to create a new ROS package from scratch and add this code & plugin configuration XML so that it builds and works.

You said something similar about pluginlib usage - it's true that it's cross-cutting but also most people who just want to get things done can't know about every possible piece of the puzzle. I think it took me 2 months to get pluginlib exports working.